pyomo/contrib/pynumero/interfaces/pyomo_grey_box_nlp.py

```diff
@@ -364,11 +364,22 @@ def get_constraints_scaling(self):
     # overloaded from NLP
     def evaluate_objective(self):
         # objective is owned by the pyomo model
-        return self._pyomo_nlp.evaluate_objective()
+        obj = 0
+        for nlp in self._nlps:
+            obj += nlp.evaluate_objective()
+        return obj
 
     # overloaded from NLP
     def evaluate_grad_objective(self, out=None):
-        return self._pyomo_nlp.evaluate_grad_objective(out=out)
+        ret = np.zeros(self.n_primals(), dtype=float)
+        for nlp in self._nlps:
+            ret += nlp.evaluate_grad_objective()
+
+        if out is not None:
+            ret.copyto(out)
+            return out
+
+        return ret
```

Comment:
Do we want to sum multiple objectives? At first thought, it seems like we should only allow one objective to be implemented. If each grey box intends to implement an objective term, maybe the methods on `ExternalGreyBoxModel` should be called e.g. `evaluate_objective_term`.

Reply:
I agree with Robby in principle that I don't always want to sum my objectives.  But am happy to accept this as a first pass